lib/cog/models/ecto_json.ex

```diff
@@ -52,6 +52,37 @@ defmodule Cog.Models.EctoJson do
     end
   end
 
+  def render_permissions(model, opts \\ []) do
+    opts = Keyword.merge(@render_defaults, opts)
+
+    policy = Keyword.fetch!(opts, :policy)
+    data = do_render(model, policy)
+    |> format_permissions(model)
+
+    case Keyword.fetch!(opts, :envelope) do
+      nil ->
+        data
+      envelope ->
+        Map.put(%{}, envelope, data)
+    end
+  end
+
+  def format_permissions(data, []), do: data
+  def format_permissions(data, %{__struct__: Cog.Models.Role}), do: data
+  def format_permissions(data, model) when is_list(model) do
+    [item | _] = model
```

Comment:
Might be cool to move this into the function head so you can match a list with at least one element in one go:

```
def format_permissions(data, [item | _]),
  do: format_permissions(data, item)
```